kong/plugins/basic-auth/daos.lua

```diff
@@ -26,4 +27,20 @@ function BasicAuthCredentials:new(properties)
   BasicAuthCredentials.super.new(self, properties)
 end
 
+function BasicAuthCredentials:insert(params)
+  if params.password then
+    return BasicAuthCredentials.super.insert(self, params)
+  else
+    -- No password was provided, so we insert a random generated password
+    local newpwd = utils.random_string()
+    params.password = newpwd
+    local data, err = BasicAuthCredentials.super.insert(self, params)
+    -- inserting the data has encrypted the password field by now, 
+    -- so add a new field with the generated plain text password
+    -- which will be returned to the requester
+    if data then data.plain_password = newpwd end
+    return data, err
+  end
+end
```

Comment:
The idea is to rather implement this at the schema level, to avoid having to do the same operations for all future DAO implementations. Kong's "Model" layer is the schema definition of an entity. It is where most of the logic should be implemented so that it is abstracted as much as possible from DAOs (and the Admin API).

Granted, the schema and the DAO are in the same file as of now, but ideally such logic should be implemented above (see the `encrypt_password` method), and leverage the schema `default` and `func`, or again `on_insert` functions to generate a random password. Things might be a bit tricky because the schema validation will not like an unknown field (plain_password), but I believe this is where the changes must be done and improvements to the schema_validation must be done too.